Custom cell and core data

CatalinDavid17 opened the pull request.

Commit:
- Partial Implementation :
	- Added: - dateLabel, noteLabel, priorityLabel and according constraints to them
		       - added the logic for "simple cells" and cell + date
		       - set selected to none for reminderCell
		       - added awakeFromNib - (additional setupt to cells)

Commit:
Priority implementation for configuring custom cell according to the data it needs.
	- tested for simple cell
	- tested for simple cell + date
	- tested for simple cell + note
	- tested for simple cell + date + note
	- tested for all of the above + priority attribute

Commit:
Fixed Constraints - made them all lower than 1000 -> not required
			      - this way it makes them flexible to priority changes

Commit:
Added dissmiss keyboard on draggin
	- removed code for dismiss on drag
	- added dismiss on drag from Interface Builder

Misantrofia commented on ReminderApp-OBJC/ReminderApp-OBJC/Classes/ReminderCell/CDReminderCell.h:
```diff
@@ -20,8 +20,18 @@
+
 @property (nonatomic, weak) IBOutlet UITextView *textView;
 @property (nonatomic, weak) id <CDReminderCellDelegate> delegate;
+@property (weak, nonatomic) IBOutlet UILabel *date;
```
Follow the naming convention, dateLabel, noteLabel, priorityLabel.

Misantrofia commented on ReminderApp-OBJC/ReminderApp-OBJC/Classes/ReminderCell/CDReminderCell.m:
```diff
@@ -17,14 +17,163 @@ @interface CDReminderCell () <UITextViewDelegate>
+
+#pragma mark Change Outlet Priorities Methods
+
+-(void)changePriorities:(CDReminder *)reminder {
```
Rename this method changePrioritiesForReminder

Misantrofia commented on ReminderApp-OBJC/ReminderApp-OBJC/Classes/ReminderCell/CDReminderCell.m:
```diff
@@ -17,14 +17,163 @@ @interface CDReminderCell () <UITextViewDelegate>
+-(void)changePriorities:(CDReminder *)reminder {
+	
+	if (!reminder.taskDate && !reminder.note &&
+		[reminder.priority isEqualToNumber:[NSNumber numberWithInteger:0]]) {
```
Use literals instead on obj C old notation "[NSNumber numberWithInteger:0]"

Misantrofia commented on ReminderApp-OBJC/ReminderApp-OBJC/Classes/ReminderCell/CDReminderCell.m:
```diff
@@ -17,14 +17,163 @@ @interface CDReminderCell () <UITextViewDelegate>
+
+#pragma mark - Helper method for converting Priority attribute into string
+
+- (NSString *)stringFromNSNumberPriority:(NSNumber *)priorityNumber {
```
Nu suna bine numele metodey, nu prea intelegi ce iese de acolo. Mai bine ceva de genul asta "priorityStringRepresentationForNumberValue"

Misantrofia commented on ReminderApp-OBJC/ReminderApp-OBJC/Classes/ReminderCell/CDReminderCell.m:
```diff
@@ -17,14 +17,163 @@ @interface CDReminderCell () <UITextViewDelegate>
+
+- (NSString *)stringFromNSNumberPriority:(NSNumber *)priorityNumber {
+	
+	switch (priorityNumber.integerValue) {
```
In Reminder "@property (nullable, nonatomic, retain) NSNumber *priority;" should be a scalar(Int_16) and you should create a NS_ENUM for this in that class. Priority Low, Medium, High, Critical and use those instead of 0, 1, 2, 3

Commit:
- Modified date, note, and priority labels name into dateLabel, noteLabel, priorityLabel
- Added NS_ENUM (CDPriorityLow, CDPriorityMedium, CDPriorityHigh, CDPriorityCritical)
- Added an external enum which gives the string according to the enum element
- modified changeProperties - changePropertiesForReminder
- modified [NSNumber numberWithInteger] with proper use of NS_ENUM elements
